python/legate_dataframe/lib/core/scalar.pyx

```diff
@@ -1,42 +1,49 @@
-# Copyright (c) 2023-2024, NVIDIA CORPORATION. All rights reserved.
+# Copyright (c) 2023-2025, NVIDIA CORPORATION. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
 # distutils: language = c++
 # cython: language_level=3
 
-from cython.operator cimport dereference
-
-from cudf._lib.scalar cimport DeviceScalar
-
 import numbers
 
 import cudf
 import legate.core
 import numpy
 
-ScalarLike = numpy.number | numbers.Number | cudf.Scalar | legate.core.Scalar
+from cudf._lib.scalar cimport DeviceScalar
+
+from legate_dataframe.lib.core.column cimport LogicalColumn
+
+ScalarLike = (
+    numpy.number | numbers.Number | cudf.Scalar | legate.core.Scalar | DeviceScalar
+)
 
 
-cdef cpp_ScalarArg cpp_scalar_arg_from_python(scalar: ScalarLike):
-    """Convert any supported Python scalar to a scalar task argument
+cdef LogicalColumn cpp_scalar_col_from_python(scalar: ScalarLike):
```

Comment:
This is very forgiving to allow conversion from all of these...

I suspect we should just disallow it eventually.  I also still relies on the non `pylibcudf` version for Python facing things (I am not sure if I can transition on cudf 24.12).

(That probably also means changing what `to_cudf()` and `to_cudf_scalar()` return, which is breaking, but seems OK.  Changing what they accept is a bit more churn.)